File: src/Entity/Abonnement.php
<?php

namespace App\Entity;

use App\Repository\AbonnementRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: AbonnementRepository::class)]
class Abonnement
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    private ?string $typeAb = null;

    #[ORM\Column]
    private ?float $prixAb = null;

    #[ORM\Column(length: 255)]
    private ?string $modePaiementAb = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTypeAb(): ?string
    {
        return $this->typeAb;
    }

    public function setTypeAb(string $typeAb): self
    {
        $this->typeAb = $typeAb;

        return $this;
    }

    public function getPrixAb(): ?float
    {
        return $this->prixAb;
    }

    public function setPrixAb(float $prixAb): self
    {
        $this->prixAb = $prixAb;

        return $this;
    }

    public function getModePaiementAb(): ?string
    {
        return $this->modePaiementAb;
    }

    public function setModePaiementAb(string $modePaiementAb): self
    {
        $this->modePaiementAb = $modePaiementAb;

        return $this;
    }
}

File: src/tn/gestion/evenement/enitite/Abonnement.java
package tn.gestion.evenement.enitite;

public class Abonnement {
    private int id;
    private String typeAb;
    private String modePaiementAb;
    private String prixAb;

    public Abonnement() {
    }

    public Abonnement(int id, String typeAb, String modePaiementAb, String prixAb) {
        this.id = id;
        this.typeAb = typeAb;
        this.modePaiementAb = modePaiementAb;
        this.prixAb = prixAb;
    }

    public Abonnement(String typeAb, String modePaiementAb, String prixAb) {
        this.typeAb = typeAb;
        this.modePaiementAb = modePaiementAb;
        this.prixAb = prixAb;
    }

    public String getPrixAb() {
        return prixAb;
    }

    public void setPrixAb(String prixAb) {
        this.prixAb = prixAb;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTypeAb() {
        return typeAb;
    }

    public void setTypeAb(String typeAb) {
        this.typeAb = typeAb;
    }

    public String getModePaiementAb() {
        return modePaiementAb;
    }

    public void setModePaiementAb(String modePaiementAb) {
        this.modePaiementAb = modePaiementAb;
    }

    @Override
    public String toString() {
        return "Categorie{" + "idCategorie=" + id + ", nom=" + typeAb + ", descriptif=" + modePaiementAb + '}';
    }
    
    
}
